core/src/main/scala/org/apache/spark/api/r/RRDD.scala

```diff
@@ -127,6 +127,13 @@ private[r] object RRDD {
       sparkConf.setExecutorEnv(name.toString, value.toString)
     }
 
+    if (sparkEnvirMap.containsKey("spark.sql.default.derby.dir") &&
+        System.getProperty("derby.system.home") == null) {
+      // This must be set before SparkContext is instantiated.
+      System.setProperty("derby.system.home",
+                         sparkEnvirMap.get("spark.sql.default.derby.dir").toString)
```

Comment:
It sounds like only R needs it. Not sure whether we should treat it as an internal `SQLConf`. 

If we do not want to put it into `SQLConf`, we need to define it in an Object, instead of hard-coding the string in all the places.